### PR TITLE
Do not change case of environment variable names

### PR DESCRIPTION
**Pull request by nuclon**
Configo makes all environment variable names uppercased. In some cases its unacceptable.

### CODE REVIEW

**Comment by bsideup** [comment]
Hi @nuclon,

First of all, thanks for your PR :)

However, I would rather make it configurable. Uppercase was implemented because of this scenario:
Source 1: `MY_PROP: 123`
Source 2: `{"my": { "prop": 456 } }`

it should resolve to `456`, but without the uppercasing it will be `123`

**Comment by nuclon** [comment]
I think in your case source 2 is resolved to "my_prop", which is correct as for me.
But I agree that it should be configurable. Environment variable CONFIGO_CASE_SENSITIVE sounds okay for you?

**Comment by bsideup** [comment]
@nuclon CONFIGO_UPPERCASE_KEYS?

**Comment by nuclon** [comment]
Ok, but I think it's reasonable to keep current behaviour as default, so this variable should be set to 0 to override. Not sure it is good idea

**Comment by bsideup** [comment]
@nuclon absolutely.

Also could you please add an integration test? I feel sorry for a lack of the docs how to contribute, my bad, will document the process soon.


Thanks!

**Comment by bsideup** [comment]
hey @nuclon, 
The change looks almost perfect. Only one thing I would like to change - do not pass the variable around, but put it to the constant near to flatten and use it there.

Motivation - users could use their own CompositeSources, and the code will fail because it's a property per CompositeSource, but it shouldn't be.

Could you please change it? Thanks!

File: sources/composite.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 
 type CompositeSource struct {
 	Sources []map[string]interface{} `json:"sources"`
+	UppercaseKeys bool
```
Comment (bsideup) [inline sources/composite.go]: I think it's unused now

**Comment by bsideup** [comment]
@nuclon, all good, thanks for your change! I'll release it soon :)

**Pull request merged**